wirecard/oxid-ee
# WDPD-85: Add Cancel/Capture/Refund UI to Transaction Details

### This PR

* Adds a new *Possible post-processing actions* tab to the transactions list detail view
* Adds hooks to cancel, capture and refund transactions to the view

### TODO

* Only display supported actions for the transaction (e.g. don't display "Capture" if it is not available for a transaction) – this can easily be done in the `_getPostProcessingActions` method
* Configure the hooks to actually cancel/capture/refund transactions

### How to Test

* Login to the admin
* Go to Extensions → Modules and activate *Wirecard Oxid EE Paymentgateway*
* Use a database manager to connect to the table `wdoxidee_ordertransactions` in the database
* Create one or more entries there; make sure to use a valid `OXID` order id for the field `WDOXIDEE_ORDERID`
* Go to Wirecard Transactions → Transactions and click on one of the displayed transactions to open the detail view
* Click the *Possible post-processing actions* tab
* Enter an invalid number (e.g. a word) and click any of the three action buttons – an error message should appear
* Enter a number smaller than zero or greater than the amount of the transaction, then click any of the three action buttons – an error message should appear
* Enter a valid number and click any of the three action buttons – a success message should appear

### Jira Links

* WDPD-85: https://jira.parkside.at/browse/WDPD-85
